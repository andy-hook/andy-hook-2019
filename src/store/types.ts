import { Action, PayloadedAction } from "./action-helpers"

export interface Store {
  loaderVisible: boolean | false
  testString: string | null
}

export interface FlushTokenAction extends Action<"auth/flush-token"> {}

export interface SetTestStringAction
  extends PayloadedAction<"set-test-string", string> {}

export interface LoaderVisibleAction
  extends PayloadedAction<"loader-visible", boolean> {}
