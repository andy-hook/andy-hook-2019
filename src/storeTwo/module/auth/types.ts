import { Action, PayloadedAction } from "../../core"

export interface AuthStore {
  token: string | null
  userId: string | null
  userName: string | null
  siteVisible: boolean | false
}

export interface SetTokenAction
  extends PayloadedAction<"auth/set-token", string> {}

export interface FlushTokenAction extends Action<"auth/flush-token"> {}

export type TokenActions = SetTokenAction | FlushTokenAction

export interface SetUserIdAction
  extends PayloadedAction<"auth/set-user-id", string> {}

export type UserIdActions = SetUserIdAction | FlushTokenAction

export interface SetUserNameAction
  extends PayloadedAction<"auth/set-user-name", string> {
  payload: string
}
export type UserNameActions = SetUserNameAction | FlushTokenAction

/* ------- */
export interface SiteVisibleAction
  extends PayloadedAction<"site-visible", boolean> {}
