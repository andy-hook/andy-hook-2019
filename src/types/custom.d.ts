// Ignore SVG imports as they are not valid to TS but supported by gatsby-plugin-react-svg
declare module "*.svg" {
  const content: any
  export default content
}

// gatsby-plugin-transition-link doesn't currently have type declarations
// https://github.com/TylerBarnes/gatsby-plugin-transition-link/issues/14
declare module "gatsby-plugin-transition-link" {
  export const TransitionState: any
  export const TransitionPortal: any
  export const Link: any

  export default Link
}

declare module "gatsby-plugin-transition-link/hooks" {
  export const useTransitionState: any
}
