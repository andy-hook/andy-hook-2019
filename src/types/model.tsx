export interface SocialItem {
  node: {
    label: string
    url: string
  }
}

export interface SocialData {
  edges: SocialItem[]
}
