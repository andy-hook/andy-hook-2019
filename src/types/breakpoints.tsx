export type TBreakpointName =
  | "bottomThumb"
  | "topThumb"
  | "bottomPalm"
  | "topPalm"
  | "bottomLap"
  | "topLap"
  | "bottomDesk"
  | "topDesk"
  | "bottomWide"
  | "topWide"
  | "bottomWall"
  | "topWall"
  | "bottomUltra"
  | "topUltra"

export type TBreakpoints = { [key in TBreakpointName]: string }
