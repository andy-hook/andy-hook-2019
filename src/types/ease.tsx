export type TEaseName =
  | "easeInCubic"
  | "easeOutCubic"
  | "easeInOutCubic"
  | "easeInCirc"
  | "easeOutCirc"
  | "easeInOutCirc"
  | "easeInExpo"
  | "easeOutExpo"
  | "easeInOutExpo"
  | "easeInQuad"
  | "easeOutQuad"
  | "easeInOutQuad"
  | "easeInQuart"
  | "easeOutQuart"
  | "easeInOutQuart"
  | "easeInQuint"
  | "easeOutQuint"
  | "easeInOutQuint"
  | "easeInSine"
  | "easeOutSine"
  | "easeInOutSine"
  | "easeInBack"
  | "easeOutBack"
  | "easeInOutBack"

export type TEases = { [key in TEaseName]: number[] }
