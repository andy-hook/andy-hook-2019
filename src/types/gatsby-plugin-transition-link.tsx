// Quick, dirty and potted types for gatsby-plugin-transition-link
interface ItransitionProps {
  delay: number
  length: number
  state: string
}

export interface ItransitionState {
  transitionStatus: "entering" | "entered" | "exiting" | "exited" | "POP"
  current: ItransitionProps
  entry: ItransitionProps
  exit: ItransitionProps
  mount: boolean
}
