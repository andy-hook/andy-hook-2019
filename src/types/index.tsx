export interface SocialItem {
  node: {
    label: string
    url: string
  }
}

export interface SocialData {
  edges: SocialItem[]
}

export interface SiteItems {
  node: {
    title: string
    description: string
    author: string
  }
}

export interface SiteData {
  allSiteJson: {
    edges: SiteItems[]
  }
}
