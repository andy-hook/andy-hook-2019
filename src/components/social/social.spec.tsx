import React from "react"
import renderer from "react-test-renderer"

import Social from "./social"
import { SocialMeta } from "../../types/model"

const mockItems: SocialMeta = {
  twitter: {
    label: "twitter",
    url: "path/to/social/profile",
  },
  instagram: {
    label: "instagram",
    url: "path/to/social/profile",
  },
  linkedin: {
    label: "linkedin",
    url: "path/to/social/profile",
  },
  dribbble: {
    label: "dribbble",
    url: "path/to/social/profile",
  },
  github: {
    label: "github",
    url: "path/to/social/profile",
  },
}

describe("<Social />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Social items={mockItems} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
