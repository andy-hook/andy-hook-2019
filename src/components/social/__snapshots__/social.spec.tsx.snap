// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Social /> renders correctly 1`] = `
<div
  className="sc-htpNat ekPird"
>
  <a
    aria-label="Twitter"
    className="sc-bxivhb gcaUoY"
    href="path/to/social/profile"
    target="_blank"
  >
    <div
      className="sc-ifAKCX XzGje sc-bdVaJa dqBkPn"
    >
      <svg
        className="icon sc-bwzfXH kdIDyY"
      >
        <use
          xlinkHref="/icon-sprite.svg#twitter"
        />
      </svg>
    </div>
  </a>
  <a
    aria-label="Instagram"
    className="sc-bxivhb gcaUoY"
    href="path/to/social/profile"
    target="_blank"
  >
    <div
      className="sc-ifAKCX XzGje sc-bdVaJa dqBkPn"
    >
      <svg
        className="icon sc-bwzfXH kdIDyY"
      >
        <use
          xlinkHref="/icon-sprite.svg#instagram"
        />
      </svg>
    </div>
  </a>
  <a
    aria-label="Linkedin"
    className="sc-bxivhb gcaUoY"
    href="path/to/social/profile"
    target="_blank"
  >
    <div
      className="sc-ifAKCX XzGje sc-bdVaJa dqBkPn"
    >
      <svg
        className="icon sc-bwzfXH kdIDyY"
      >
        <use
          xlinkHref="/icon-sprite.svg#linkedin"
        />
      </svg>
    </div>
  </a>
  <a
    aria-label="Dribbble"
    className="sc-bxivhb gcaUoY"
    href="path/to/social/profile"
    target="_blank"
  >
    <div
      className="sc-ifAKCX XzGje sc-bdVaJa dqBkPn"
    >
      <svg
        className="icon sc-bwzfXH kdIDyY"
      >
        <use
          xlinkHref="/icon-sprite.svg#dribbble"
        />
      </svg>
    </div>
  </a>
  <a
    aria-label="Github"
    className="sc-bxivhb gcaUoY"
    href="path/to/social/profile"
    target="_blank"
  >
    <div
      className="sc-ifAKCX XzGje sc-bdVaJa dqBkPn"
    >
      <svg
        className="icon sc-bwzfXH kdIDyY"
      >
        <use
          xlinkHref="/icon-sprite.svg#github"
        />
      </svg>
    </div>
  </a>
</div>
`;
